compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -942,7 +942,6 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         let mut bounds = Bounds::default();
 
         self.add_bounds(param_ty, ast_bounds, &mut bounds);
-        bounds.trait_bounds.sort_by_key(|(t, _, _)| t.def_id());
```

Comment:
This code appears to date back to 2014 from PR https://github.com/rust-lang/rust/pull/15985:

https://github.com/rust-lang/rust/blob/08f033b8b73a00180955096f2c03df187f9b2060/src/librustc/middle/typeck/collect.rs#L1221